server/controllers/users.js

```diff
@@ -0,0 +1,139 @@
+const User = require('../models').users;
+const jwt = require('jsonwebtoken');
+
+module.exports = {
+  create(req, res) {
+    return User
+      .create(req.body)
+      .then((user) => res.status(201).send({
+        message: 'User Created Successfully',
+        firstName: user.firstName,
+        lastName: user.lastName,
+        id: user.id,
+      }))
+      .catch(error => res.status(400).send(error.errors));
+  },
+  
+  // login(req, res) {
+  //   User
+  //     .findOne({
+  //       where: {
+  //         email: req.body.email,
+  //         password: req.body.password
+  //       }
+  //     })
+  //     .then(user => {
+  //       const token = jwt.sign(user.email, process.env.SECRET_KEY = 'superSecretString');
+  //       if (!user) {
+  //         return res.status(404).send({
+  //           message: 'user not found'
+  //         })
+  //       }
+  //       return res.status(200).send(token);
+  //     })
+  // },
+
+  list(req, res) {
+    return User
+      .findAll()
+      .then(user => res.status(200).send(user))
+      .catch(error => res.status(400).send(error));
+  },
+
+  retrieveOne(req, res) {
+    return User
+      .findById(req.params.userId)
+      .then(user => {
+        if (!user) {
+          return res.status(404).send({
+            message: 'Sorry. User not found',
+          });
+        }
+        return res.status(200).send(user);
+      })
+      .catch(error => res.status(400).send(error));
+  },
+
+  retrieveLimited(req, res) {
```

Comment:
**Code Quality - Critical**

- You handle this in the `list` function where you will check whether there are any necessary queries to modify the users to return